File: components/credentials/src/main/java/org/cloudfoundry/credhub/requests/DefaultCredentialGenerateRequest.java
package org.cloudfoundry.credhub.requests;

import com.fasterxml.jackson.annotation.JsonIgnore;

public class DefaultCredentialGenerateRequest extends BaseCredentialGenerateRequest {

  private Object parameters;

  public Object getParameters() {
    return parameters;
  }

  public void setParameters(final Object parameters) {
    this.parameters = parameters;
  }

  @Override
  @JsonIgnore
  public GenerationParameters getGenerationParameters() {
    return null;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/requests/DefaultCredentialGenerateRequest.kt
package org.cloudfoundry.credhub.requests

import com.fasterxml.jackson.annotation.JsonIgnore

class DefaultCredentialGenerateRequest : BaseCredentialGenerateRequest() {

    var parameters: Any? = null

    override val generationParameters: GenerationParameters?
        @JsonIgnore
        get() = null
}
